Implement page layouts with documentation.

adunkman opened the pull request:
Closes #34.

This pull request introduces styles which apply to major page components, like:

- The “An official website of the United States government” banner
- The main logo and header, including primary and secondary navigation
- Side navigation
- Footer with GSA logo

The documentation page renders previews of these via zoomed out `<iframe>`s. I decided to skip embedding code samples based on my experience using the U.S. Web Design System’s page layouts (it’s a lot faster to just inspect the source to jump to the right section of the layout, since the code sample is so long). 

Here’s what it all looks like — and feel free to click through to the Federalist preview: 

[![Screenshot of layout documentation page](https://user-images.githubusercontent.com/14930/52133233-eed75d80-260e-11e9-8852-23ad3cc4612b.png)](https://federalist-proxy.app.cloud.gov/preview/18f/identity-style-guide/layouts/layout/)

adunkman commented:
Interesting, and nice catch! I’ll take a look (as per the Sketch, they should not reach the edges on internal items)!

donjo commented:
Ahh I see the problem. The imported sidenav is using `usa-nav-primary-item` on the list items in the nav but it should be using `usa-sidenav-item` per the examples

adunkman commented:
Fixed the navigation helpers for the style guide to appropriately apply sidenav classes in e5afb48, and followed-up with 1058054 to fix the styles to apply to the sidenav correctly. 

Additionally, rebased to `master` to fix merge conflicts. 

Thanks for catching that! I believe we should be good to go — mind giving it a double-check? :heart:
